src/libstore/globals.hh

```diff
@@ -966,6 +984,23 @@ public:
 
     Setting<bool> acceptFlakeConfig{this, false, "accept-flake-config",
         "Whether to accept nix configuration from a flake without prompting."};
+
+    EnvironmentSetting environment{this, {}, "environment",
+        R"(
+          Extra environments variables to use. A list of items, each in the
+          format of:
+
+          - `name=value`: Set environment variable `name` to `value`.
+          - `name`: Inherit environment variable `name` from current
+            environment.
+
+          If the user is trusted (see `trusted-users` option), the daemon will
+          also have these environment variables set.
+
+          This option is useful for, e.g., setting `https_proxy` for
+          fixed-output derivations and substituter downloads in a multi-user
+          Nix installation.
+        )"};
```

Comment:
I think this is missing a warning that it could potentially break things. For example what happens if I would do a typo and overwrite important envs like HOME, TMP or NIX_STORE?

Reply:
A 'typo' here honestly feels unlikely, but I agree that a warning that this is potentially breaking.